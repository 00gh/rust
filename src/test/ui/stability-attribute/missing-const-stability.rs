#![feature(staged_api)]
#![stable(feature = "stable", since = "1.0.0")]

#[stable(feature = "stable", since = "1.0.0")]
pub const fn foo() {} //~ ERROR function has missing const stability attribute

#[unstable(feature = "unstable", issue = "none")]
pub const fn bar() {} // ok for now

#[stable(feature = "stable", since = "1.0.0")]
pub struct Foo;
impl Foo {
    #[stable(feature = "stable", since = "1.0.0")]
    pub const fn foo() {} //~ ERROR associated function has missing const stability attribute

    #[unstable(feature = "unstable", issue = "none")]
    pub const fn bar() {} // ok for now
}

// FIXME When #![feature(const_trait_impl)] is stabilized, add tests for const
// trait impls. Right now, a "trait methods cannot be stable const fn" error is
// emitted, but that's not in the scope of this test.

fn main() {}
