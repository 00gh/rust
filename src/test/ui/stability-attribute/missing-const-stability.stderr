error: function has missing const stability attribute
  --> $DIR/missing-const-stability.rs:5:1
   |
LL | pub const fn foo() {}
   | ^^^^^^^^^^^^^^^^^^^^^

error: associated function has missing const stability attribute
  --> $DIR/missing-const-stability.rs:14:5
   |
LL |     pub const fn foo() {}
   |     ^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

